mlflow/server/js/src/experiment-tracking/components/ExperimentListView.js

```diff
@@ -86,76 +121,130 @@ export class ExperimentListView extends Component {
     this.updateSelectedExperiment('0', '');
   };
 
-  handleCheck = (checkedKeys) => {
-    if (checkedKeys.length > 0) {
+  // Add a key if it does not exist, remove it if it does
+  // Always keep at least one experiment checked if it is only the active one.
+  handleCheck = (e, key) => {
+    this.setState((prevState, props) => {
+      let { checkedKeys } = prevState;
+      if (e === true && !prevState.checkedKeys.includes(key)) {
+        checkedKeys = [key, ...prevState.checkedKeys];
+      }
+      if (e === false && props.activeExperimentIds.length !== 1) {
+        checkedKeys = prevState.checkedKeys.filter((i) => i !== key);
+      }
+      return { checkedKeys: checkedKeys };
+    }, this.pushExperimentRoute);
+  };
+
+  pushExperimentRoute = () => {
+    if (this.state.checkedKeys.length > 0) {
       const route =
-        checkedKeys.length === 1
-          ? Routes.getExperimentPageRoute(checkedKeys[0])
-          : Routes.getCompareExperimentsPageRoute(checkedKeys);
+        this.state.checkedKeys.length === 1
+          ? Routes.getExperimentPageRoute(this.state.checkedKeys[0])
+          : Routes.getCompareExperimentsPageRoute(this.state.checkedKeys);
       this.props.history.push(route);
     }
   };
 
-  renderListItem = ({ title, key }) => {
+  onScroll = ({ startIndex, stopIndex }) => {
+    const isScrolledToLastItem = stopIndex >= this.props.experiments.length;
+    if (!isScrolledToLastItem) {
+      return;
+    }
+  };
+
+  // Avoid calling emotion for every list item
+  activeExperimentListItem = classNames.getExperimentListItemContainer(
+    true,
+    this.props.designSystemThemeApi.theme,
+  );
+  inactiveExperimentListItem = classNames.getExperimentListItemContainer(
+    false,
+    this.props.designSystemThemeApi.theme,
+  );
+
+  renderListItem = ({ index, key, style, isScrolling }) => {
+    const item = this.state.filteredExperiments[index];
     const { activeExperimentIds } = this.props;
-    const isActive = activeExperimentIds.includes(key);
+    const { checkedKeys } = this.state;
+    const isActive = activeExperimentIds.includes(item.experiment_id);
+    const isChecked = checkedKeys.includes(item.experiment_id);
     const dataTestId = isActive ? 'active-experiment-list-item' : 'experiment-list-item';
```

Comment:
there's a slight regression bug: after selecting (checking) multiple experiments and then navigating back in history, the checkboxes remain checked/unchecked 🤔 I guess it wasn't an issue before